src/compat/eme/set_media_keys.ts

```diff
@@ -0,0 +1,49 @@
+import log from "../../log";
+import sleep from "../../utils/sleep";
+import shouldAwaitSetMediaKeys from "../should_await_set_media_keys";
+import type { ICustomMediaKeys } from "./custom_media_keys";
+import type { IEmeApiImplementation } from "./eme-api-implementation";
+
+/**
+ * @param {Object} emeImplementation
+ * @param {Object} mediaElement
+ * @param {Object|null} mediaKeys
+ * @returns {Promise}
+ */
+export function setMediaKeys(
+  emeImplementation: IEmeApiImplementation,
+  mediaElement: HTMLMediaElement,
+  mediaKeys: MediaKeys | ICustomMediaKeys | null,
+): Promise<unknown> {
+  const prom = emeImplementation
+    .setMediaKeys(mediaElement, mediaKeys)
+    .then(() => {
+      log.info("Compat: MediaKeys updated with success");
+    })
+    .catch((err) => {
+      if (mediaKeys === null) {
+        log.error(
+          "Compat: Could not reset MediaKeys",
+          err instanceof Error ? err : "Unknown Error",
+        );
+        return;
+      }
+      log.error(
+        "Compat: Could not update MediaKeys",
+        err instanceof Error ? err : "Unknown Error",
+      );
+      throw err;
+    });
+
+  if (shouldAwaitSetMediaKeys()) {
+    return prom;
+  }
+
+  return Promise.race([
+    prom,
+
+    // Because we know how much EME has implementation issues, let's not block
+    // everything because that API hangs
+    sleep(1000),
+  ]);
```

Comment:
We need to determine the approach to address this issue: should we await for `setMediaKeys` or not?
Should there be an exception for devices that require waiting, such as the LG TV,
or should there be an exception for devices that should not wait because they hangs (though we're uncertain which devices fall into this category)?